### DISCUSSION TITLE
Proposal: Module-publishing GitHub app

### DISCUSSION THREAD

**Discussion by rogpeppe** [cue-lang/cue#2448] [Proposal]
Proposal #2330 presented a updated modules proposal. This discussion is for a sub-proposal of that: details on how we propose to initially to use a GitHub app to upload modules.

See here for the sub-proposal text in full: https://github.com/cue-lang/proposal/blob/main/designs/modules/2448-modules-github.md

**Comment by verdverm** [comment]
Why not an action over an app?

Apps require approvals and installation and are generally more opaque in how they work. Each new registry will require approval before a developer can use it, assuming that admins will assign the registries that are accessibly, over granting access to all repositories.

Actions can be included in existing workflows without requiring admin approval.

**Comment by verdverm** [comment]
What exactly is the issue with existing tags that requires this extra `cue-*` tag?

**Comment by myitcv** [comment]
This proposal was superseded by #2939. As we updated in https://github.com/cue-lang/cue/discussions/2939#discussioncomment-9468945, that proposal (and its sub-proposals) have now been accepted, hence we are closing this discussion to represent the decision on the v3 proposal.
